junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ClassBasedTestDescriptor.java

```diff
@@ -139,6 +146,22 @@ public final String getLegacyReportingName() {
 		return this.testClass.getName();
 	}
 
+	// --- Validatable ---------------------------------------------------------
+
+	@Override
+	public void validate(IssueReporter reporter) {
+		beforeAllMethods.stream() //
+				.filter(ModifierSupport::isPrivate) //
+				.forEach(method -> reporter.reportIssue(Severity.DEPRECATION,
+					String.format("@BeforeAll method [%s] should not be private.", method.toGenericString()),
+					b -> b.source(MethodSource.from(method))));
+		afterAllMethods.stream() //
+				.filter(ModifierSupport::isPrivate) //
+				.forEach(method -> reporter.reportIssue(Severity.DEPRECATION,
+					String.format("@AfterAll method [%s] should not be private.", method.toGenericString()),
+					b -> b.source(MethodSource.from(method))));
```

Comment:
:speech_balloon: Example of a engine-specific "post-discovery" validation that reports private lifecycle methods as deprecated (also would need to be done for `@BeforeEach` etc.).